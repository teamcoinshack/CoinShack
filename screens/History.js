import React, {Component} from 'react';
import {
  Text, 
  View, 
  Dimensions,
  StyleSheet, 
  FlatList, 
  ScrollView,
  Alert,
  RefreshControl,
} from 'react-native';
import Firebase from 'firebase';
import { LoginManager, AccessToken } from 'react-native-fbsdk'
import db from '../Database.js';

const background = '#373b48';

export default class History extends Component {
  constructor(props) {
    super(props);

    this.state = {
      history: [] 
    }
    this.renderRow = this.renderRow.bind(this);
  }

  async componentDidMount() {
    try {
      const uid = Firebase.auth().currentUser.uid;
      const snap = await db.getData(uid);
      this.setState({
        history: snap.val().history,
      })
      console.log(snap.val().history);
    } catch(error) {
      console.log(error);
    }
  }

  renderRow({item}) {
    let d = new Date(item.date);
    const day = d.getDay();
    const month = d.getMonth();
    const year = d.getFullYear();

    let hour = d.getHours();
    const min = d.getMinutes();
    const postfix = hour >= 12 ? ' PM' : ' AM';
    hour = hour > 13 ? hour - 12 : hour;

    const dateString = 
      day + '/' + month + '/' + year + ' - ' + hour + ':' + min + postfix;
    return (
      <View
        style={styles.row}
      >
        <View style={{
          flexDirection: 'row',
          flex: 1,
        }}>
          <View style={{
            flexDirection: 'column',
            alignItems: 'flex-start',
          }}>
            {item.buy 
              ? ( <Text style={styles.buy}>BUY</Text> )
              : ( <Text style={styles.sell}>SELL</Text> ) }
            <Text style={styles.row2}>
              {db.stringify(item.coinValue.toFixed(5)) 
                + ' ' + item.symbol + ' @ $' + db.stringify(item.rate)}
            </Text>
            <Text style={styles.row3}>
              {dateString}
            </Text>
          </View>
        </View>
      </View>
    )
  }

  render() {
    return (
      <View style={styles.container}>
        <FlatList
          style={styles.flatStyle}
          data={this.state.history}
          renderItem={this.renderRow}
          keyExtractor={item => item.date}
        />
      </View>
    )
  }
}

const styles = StyleSheet.create({
  flatStyle: {
    marginTop: 10,
  },
  container: {
    flex: 1,
    alignSelf: "stretch",
    backgroundColor: background,
  },
  row: {
    elevation: 1,
    borderRadius: 5,
    backgroundColor: '#515360',
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'flex-start',
    alignItems: 'center',
    paddingTop: 15,
    paddingBottom: 15,
    paddingLeft: 18,
    paddingRight: 16,
    marginLeft: 14,
    marginRight: 14,
    marginTop: 0,
    marginBottom: 6,
  },
  buy: {
    color: '#14ff81', 
    textAlignVertical: 'bottom',
    includeFontPadding: false,
    flex: 0,
    fontSize: 20,
    fontWeight: '600',
  },
  sell: {
    color: '#ff077a', 
    textAlignVertical: 'bottom',
    includeFontPadding: false,
    flex: 0,
    fontSize: 20,
    fontWeight: '600',
  },
  row2: {
    color: '#dbdbdb',
    fontWeight: '600',
    fontSize: 18,
    paddingTop: 5,
  },
  row3: {
    color: '#a4a9b9',
    fontWeight: '500',
    paddingTop: 5,
    fontSize: 15,
  }
});
