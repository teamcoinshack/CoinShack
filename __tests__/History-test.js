/**
 * @format
 */

import 'react-native';
import React from 'react';
import History from '../screens/History.js';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

jest.mock('react-native-google-signin', () => {});
test('renders correctly', () => {
  const tree = renderer.create(<History />).toJSON();
  expect(tree).toMatchSnapshot();
});
