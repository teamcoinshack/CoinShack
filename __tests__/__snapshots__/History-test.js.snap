// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignSelf": "stretch",
      "backgroundColor": "#373b48",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "height": 65,
          "justifyContent": "center",
          "width": 525,
        }
      }
    >
      <View
        animated={true}
        animationConfig={
          Object {
            "bounciness": 0,
          }
        }
        animationType="spring"
        indeterminate={true}
        indeterminateAnimationDuration={1000}
        onLayout={[Function]}
        progress={0.3}
        style={
          Array [
            Object {
              "backgroundColor": undefined,
              "borderColor": "#80e7a5",
              "borderRadius": 4,
              "borderWidth": 0,
              "overflow": "hidden",
              "width": 250,
            },
            undefined,
          ]
        }
        useNativeDriver={false}
      >
        <View
          style={
            Object {
              "backgroundColor": "#80e7a5",
              "height": 6,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
                Object {
                  "translateX": -87.5,
                },
                Object {
                  "scaleX": 0.30007,
                },
              ],
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;
